src/QMCDrivers/WFOpt/QMCCostFunctionBatched.cpp

```diff
@@ -385,12 +365,9 @@ void QMCCostFunctionBatched::checkConfigurations(EngineHandle& handle)
           RecordsOnNode[is][REWEIGHT]     = 1.0;
           RecordsOnNode[is][ENERGY_FIXED] = h_list[ib].getLocalPotential();
 
-          if (includeNonlocalH != "no")
-          {
-            OperatorBase* nlpp = h_list[ib].getHamiltonian(includeNonlocalH);
-            if (nlpp)
-              RecordsOnNode[is][ENERGY_FIXED] -= nlpp->getValue();
-          }
+          const auto twf_dependent_components = h_list[ib].getTWFDependentComponents();
+          for (const OperatorBase& component: twf_dependent_components)
+            RecordsOnNode[is][ENERGY_FIXED] -= component.getValue();
```

Comment:
Previously, this value was only included if nlpp was set.  Now it checks for TWF-dependent components, which includes the kinetic energy.